internal/future/app/mmlogic/mmlogic_service.go

```diff
@@ -43,21 +55,51 @@ func newMmlogic(cfg config.View) (*mmlogicService, error) {
 	return ms, nil
 }
 
-// GetPoolTickets gets the list of Tickets that match every Filter in the
+// QueryTickets gets the list of Tickets that match every Filter in the
 // specified Pool.
-func (s *mmlogicService) QueryTickets(req *pb.QueryTicketsRequest, stream pb.MmLogic_QueryTicketsServer) error {
-	ctx := stream.Context()
+func (s *mmlogicService) QueryTickets(req *pb.QueryTicketsRequest, responseServer pb.MmLogic_QueryTicketsServer) error {
+	poolFilters := req.Pool.Filter
+
+	// Send requests to the storage service
+	idsToProperties, err := s.store.FilterTickets(poolFilters)
+	if err != nil {
+		logger.WithError(err).Error("Failed to retrieve result from storage service.")
+		return status.Error(codes.Internal, err.Error())
+	}
 
-	for {
+	ctx := responseServer.Context()
+	idx := 0
+	// The number of tickets in a paging response
+	pSize := s.cfg.GetInt("storage.page.size")
+	page := []*pb.Ticket{}
+
+	for id, property := range idsToProperties {
```

Comment:
There is some paging logic in this that is interesting to test. May be break this helper into a separate method - so you can give it a populated map and validate it handles paging properly?

Reply:
I will add the paging logic tests to `redis_test.go` file later. I didn't do this for now because I want to confirm the functions and signatures are well-established to avoid refactoring too many files

Reply:
Ok. Create an issue to track the paging behavior / validation of the same. I think we need some thought to how excess results will be handled through the stack (in mmlogic / at storage layer)